pkg/components/rook/component_conversion_test.go

```diff
@@ -121,107 +121,112 @@ component "rook" {
+		"storage_node_key2": "storage_node_value2",
+	}
+	got := deploy.Spec.Template.Spec.NodeSelector
+
+	if !reflect.DeepEqual(expected, got) {
+		t.Fatalf("Expected: %v, got: %v", expected, got)
+	}
+}
+
+func monitoring(t *testing.T, m map[string]string, deploy *appsv1.Deployment) {
+	expectedConfig := []string{
+		"rook-ceph/templates/ceph-cluster.yaml",
+		"rook-ceph/templates/ceph-osd.yaml",
+		"rook-ceph/templates/ceph-pools.yaml",
+		"rook-ceph/templates/csi-metrics-service-monitor.yaml",
+		"rook-ceph/templates/prometheus-ceph-v14-rules.yaml",
+		"rook-ceph/templates/service-monitor.yaml",
+	}
+
+	for _, ec := range expectedConfig {
+		if _, ok := m[ec]; !ok {
+			t.Fatalf("Expected %q to be generated but it is not available", ec)
+		}
+	}
+}
+
+func operatorTolerations(t *testing.T, m map[string]string, deploy *appsv1.Deployment) {
+	expected := []corev1.Toleration{
+		{Key: "toleration_key1", Operator: "Equal", Value: "toleration_value1", Effect: "NoSchedule"},
+		{Key: "toleration_key2", Operator: "Equal", Value: "toleration_value2", Effect: "NoSchedule"},
+	}
+	got := deploy.Spec.Template.Spec.Tolerations
+
+	if !reflect.DeepEqual(expected, got) {
+		t.Fatalf("Expected: %v, got: %v", expected, got)
+	}
+}
+
+func envVars(t *testing.T, m map[string]string, deploy *appsv1.Deployment) {
+	// nolint:lll
+	tcs := map[string]string{
+		"AGENT_TOLERATION_KEY":          "agent_toleration_key",
+		"AGENT_TOLERATION":              "NoSchedule",
+		"AGENT_TOLERATIONS":             "- effect: NoSchedule\n  key: toleration_key1\n  operator: Equal\n  value: toleration_value1\n- effect: NoSchedule\n  key: toleration_key2\n  operator: Equal\n  value: toleration_value2",
+		"DISCOVER_TOLERATION_KEY":       "discover_toleration_key",
+		"DISCOVER_TOLERATION":           "NoSchedule",
+		"DISCOVER_TOLERATIONS":          "- effect: NoSchedule\n  key: toleration_key1\n  operator: Equal\n  value: toleration_value1\n- effect: NoSchedule\n  key: toleration_key2\n  operator: Equal\n  value: toleration_value2",
+		"CSI_PROVISIONER_TOLERATIONS":   "- effect: NoSchedule\n  key: toleration_key1\n  operator: Equal\n  value: toleration_value1\n- effect: NoSchedule\n  key: toleration_key2\n  operator: Equal\n  value: toleration_value2",
+		"CSI_PLUGIN_TOLERATIONS":        "- effect: NoSchedule\n  key: other_node_key1\n  operator: Equal\n  value: other_node_value1",
+		"AGENT_NODE_AFFINITY":           "storage_node_key1=storage_node_value1; storage_node_key2=storage_node_value2;",
+		"DISCOVER_AGENT_NODE_AFFINITY":  "storage_node_key1=storage_node_value1; storage_node_key2=storage_node_value2;",
+		"CSI_PROVISIONER_NODE_AFFINITY": "storage_node_key1=storage_node_value1; storage_node_key2=storage_node_value2;",
+		"CSI_PLUGIN_NODE_AFFINITY":      "node_key1=node_value1; node_key2=node_value2;",
+	}
+
+	envs := deploy.Spec.Template.Spec.Containers[0].Env
+
+	// Convert the envs into map so that comparison becomes easier.
+	envVars := make(map[string]string, len(envs))
```

Comment:
Maybe this for consistent initialization of maps? I don't think we benefit much from using `make` with predefined length.
```suggestion
	envVars := map[string]string{}
```